### DISCUSSION TITLE
Save file to specific directory

### DISCUSSION THREAD

**Discussion by aravindkumarlotus** [jimmywarting/StreamSaver.js#277] [Q&A] [answered]
hi, is it possible to save files to a user-selected folder? I tried and failed using streamsaver.js. please advise how it should be handled

**Comment by jimmywarting** [comment]
Sry, but that isn't possible with StreamSaver (or FileSaver.js for that mather)
it uses the same browser tech used by server + browser when it responds with regular `content-disposition` attachment header that tells the browser to save the file. 
StreamSaver don't have control over how browser handles the saving.

This is a browser settings and it can be adjusted by the user of where it should save the files (weather or not it should auto save or let the client pick a folder)

**Comment by jimmywarting** [comment, marked as answer]
there is a new browser api available that will make this library a bit obsolete eventually when https://github.com/WICG/file-system-access becomes more accessible.

With this new file system access you can use `await window.showSaveFilePicker()` to ask the user where it should save the file and even what name or format the user wants to save it at. 
it's more like a "save as" rather than just plain "save"

you can also use `window.showDirectoryPicker()` to ask for a specific folder and ask for write permission to that folder...
then you can write data to wherever you want without prompting the user further. 

this new picker should have an option for hinting where it should save a file.
https://github.com/WICG/file-system-access/blob/main/SuggestedNameAndDir.md

**Reply by jimmywarting** [reply]
doe it's limited to just chrome...
